packages/in_app_purchase/in_app_purchase_ios/ios/Classes/FIAObjectTranslator.m

```diff
@@ -193,4 +208,34 @@ + (NSDictionary *)getMapFromSKStorefront:(SKStorefront *)storefront
   return map;
 }
 
++ (SKPaymentDiscount *)getSKPaymentDiscountFromMap:(NSDictionary *)map {
+  if (!map || map.count <= 0) {
+    return nil;
+  }
+
+  NSAssert(map[@"identifier"],
+           @"When specifying a payment discount the 'identifier' field is mandatory.");
+  NSAssert(map[@"keyIdentifier"],
+           @"When specifying a payment discount the 'keyIdentifier' field is mandatory.");
+  NSAssert(map[@"nonce"], @"When specifying a payment discount the 'nonce' field is mandatory.");
+  NSAssert(map[@"signature"],
+           @"When specifying a payment discount the 'signature' field is mandatory.");
+  NSAssert(map[@"timestamp"],
+           @"When specifying a payment discount the 'timestamp' field is mandatory.");
+
+  NSString *identifier = map[@"identifier"];
+  NSString *keyIdentifier = map[@"keyIdentifier"];
+  NSUUID *nonce = [[NSUUID alloc] initWithUUIDString:map[@"nonce"]];
```

Comment:
will `initWithUUIDString` crash if map[@"nonce"] is nil?

Reply:
According to [Apple's documentation](https://developer.apple.com/documentation/foundation/nsuuid/1411662-initwithuuidstring?language=objc) the `initWithUUIDString` method returns `nil` for invalid strings. I have made a quick test and confirmed this also includes `nil` values passed in.

I did restructure the code to improve error handling (see earlier comment).